### DISCUSSION TITLE
C# Desktop does it finally support NativeAOT?

### DISCUSSION THREAD

**Discussion by Symbai** [microsoft/microsoft-ui-xaml#8015] [Q&A] [answered]
Last time I've tried WinUI 3 it generates a C# desktop application with full JIT, just like WPF or Winforms. Which kinda surprised me as WinUI comes from UWP area where even C# gets compiled into native code. 

Has there been any changes so far? Can you compile it using nativeaot?

**Comment by pjmlp** [comment, marked as answer]
No you can't, this is yet another area where WinUI lags behind UWP.

See [Limitations of Native AOT deployment](https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/#limitations-of-native-aot-deployment)

**Reply by Symbai** [reply]
So no official Desktop UI from Microsoft which supports nativeaot, not even on paper/roadmap. That is frustrating to hear.

**Reply by YegorStepanov** [reply]
@Symbai Currently only console apps support nativeaot. Even asp.net doesn't support it (the most important topic for microsoft in C#)
